pkg/controller/acmeorders/sync_test.go

```diff
@@ -1412,6 +1412,39 @@ func TestChallengeSpecForAuthorization(t *testing.T) {
 				},
 			},
 		},
+		"uses correct solver when selector explicitly names dnsName (reversed)": {
+			acmeClient: basicACMEClient,
+			issuer: &v1alpha1.Issuer{
+				Spec: v1alpha1.IssuerSpec{
+					IssuerConfig: v1alpha1.IssuerConfig{
+						ACME: &v1alpha1.ACMEIssuer{
+							Solvers: []v1alpha1.ACMEChallengeSolver{
+								exampleComDNSNameSelectorSolver,
+								emptySelectorSolverHTTP01,
```

Comment:
Would this test have failed with the previous implementation? It seems like it would not, as when `emptySelectorSolverHTTP01` is checked, `selectedSolver` would not equal nil (from what I can tell).

Reply:
The nil would be from `*cfg.Selector` when it tries get the matching data, since selector is nil in `emptySelectorSolverHTTP01`